test/e2e/e2e_toomanyrestarts_test.go

```diff
@@ -192,7 +193,14 @@ func waitPodRestartCount(ctx context.Context, clientSet clientset.Interface, nam
 				return false, nil
 			}
 			for i := 0; i < 4; i++ {
-				if len(podList.Items[0].Status.ContainerStatuses) < 1 {
+				wait.Poll(5*time.Second, 2*time.Minute, func() (bool, error) {
+					t.Logf("Waiting for podList.Items[%v] to be scheduled , now pod status is [%v]", podList.Items[i].Name, podList.Items[i].Status.Phase)
+					if podList.Items[i].Status.Phase != v1.PodPending || podList.Items[i].Status.Phase == v1.PodRunning {
```

Comment:
it will never evaluate the second condition if it is running
```suggestion
					if podList.Items[i].Status.Phase != v1.PodPending
```

Reply:
yes, you are right, if not pending then we do not need the second condition, i will update now~

Reply:
update now, may i click `resolve conversation`?